### ISSUE TITLE
Max(x, y).is_Function and isinstance(Max(x, y), Function) are False

### ISSUE THREAD

**Issue by smichr** [sympy/sympy#14242]
Is there a reason for these functions to not appear as Function? (They are Application type.)

**Comment by jashan498** [comment]
Being `Application` type makes `Max(x, y).is_Function = True` but `Max` is not inherited from `Function` that is why  `isinstance(Max(x, y), Function) ` is False

**Comment by smichr** [comment]
Other than making Max inherit from Function, is there any way to make it so `isinstance(Max(x, 1), Function)` would be true? Is there a reason that Max doesn't inherit from Function (perhaps in addition to Application)?

**Comment by jashan498** [comment]
I don't think it would be possible until `Max` is a subclass of `Function`. I have seen a similar issue [here](https://github.com/sympy/sympy/issues/14000).

**Comment by jksuom** [comment]
Max and Min differ from most functions in that they are restricted to the real domain. I don't know if this was the reason for not making them inherit from Function, but it might not be a bad idea to keep it that way.

**Comment by normalhuman** [comment]
Heaviside is also restricted to real domain, but inherits from Function nonetheless. 

I thought that the fact that the number of arguments of Max and Min is variable could be a factor here. But does anything actually break if the definition of Max was `class Max(MinMaxBase, Function)`?
